app/src/main/java/com/example/weatherapp/ForecastActivity.kt

```diff
@@ -1,62 +1,73 @@
-        DayForecast(1644273472, 1644214904, 1644262244, forecastTempData[0], 1000f, 30),
-        DayForecast(1644359872, 1644300224, 1644348164, forecastTempData[1], 1000f, 30),
-        DayForecast(1644446272, 1644391004, 1644437804, forecastTempData[2], 1000f, 30),
-        DayForecast(1644532672, 1644475664, 1644522224, forecastTempData[3], 1000f, 30),
-        DayForecast(1644619072, 1644562244, 1644605624, forecastTempData[4], 1000f, 30),
-        DayForecast(1644705472, 1644649604, 1644692324, forecastTempData[5], 1000f, 30),
-        DayForecast(1644791872, 1644735764, 1644782564, forecastTempData[6], 1000f, 30),
-        DayForecast(1644878272, 1644819044, 1644868004, forecastTempData[7], 1000f, 30),
-        DayForecast(1644964672, 1644905744, 1644952544, forecastTempData[8], 1000f, 30),
-        DayForecast(1645051072, 1644992504, 1645039304, forecastTempData[9], 1000f, 30),
-        DayForecast(1645137472, 1645079204, 1645126064, forecastTempData[10], 1000f, 30),
-        DayForecast(1645223872, 1645166564, 1645213664, forecastTempData[11], 1000f, 30),
-        DayForecast(1645310272, 1645252724, 1645299284, forecastTempData[12], 1000f, 30),
-        DayForecast(1645396672, 1645386464, 1645383404, forecastTempData[13], 1000f, 30),
-        DayForecast(1645483072, 1645423844, 1645470764, forecastTempData[14], 1000f, 30),
-        DayForecast(1645569472, 1645511384, 1645557704, forecastTempData[15], 1000f, 30),
-    )
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        val moshi = Moshi.Builder()
+            .add(KotlinJsonAdapterFactory())
+            .build()
+
+        val retrofit = Retrofit.Builder()
+            .baseUrl("https://pro.openweathermap.org/data/2.5/forecast/")
+            .addConverterFactory(MoshiConverterFactory.create(moshi))
+            .build()
+
+        api = retrofit.create(Api::class.java)
+
         setContentView(R.layout.activity_forecast)
 
         recyclerView = findViewById(R.id.recyclerView)
 
         recyclerView.layoutManager = LinearLayoutManager(this)
 
-        recyclerView.adapter = MyAdapter(adapterData)
 
-//        forecastTempData.get(0)
+    }
+
+    override fun onResume() {
+        super.onResume()
+        println("onResume ForecastActivity")
+        val call: Call<Forecast> = api.getForecast("55124")
+        call.enqueue(object : Callback<Forecast> {
+            override fun onResponse(
+                call: Call<Forecast>,
+                response: Response<Forecast>
+            ) {
+                val forecastConditions = response.body()
+                forecastConditions?.let {
+                    bindData(it.list)
+                }
+
+            }
+
+            override fun onFailure(call: Call<Forecast>, t: Throwable) {
+                t.printStackTrace()
+            }
+        })
+    }
+
+    private fun bindData(forecast: List<DayForecast>) {
+        adapterData = forecast
```

Comment:
You don't need this line. It could just be
`recyclerView.adapter = MyAdapter(forecast)`